src/lib/internal/actions/interact-outside/action.ts

```diff
@@ -0,0 +1,96 @@
+import { addEventListener, isElement, executeCallbacks, noop } from '$lib/internal/helpers';
+import type { InteractOutsideConfig, InteractOutsideEvent } from './types.js';
+
+export function useInteractOutside(node: HTMLElement, config: InteractOutsideConfig) {
+	let unsub = noop;
+	function update(config: InteractOutsideConfig) {
+		unsub();
+		const { onInteractOutside, onInteractOutsideStart, enabled } = config;
+
+		if (!enabled) {
+			return {
```

Comment:
Why are you returning destroy here? You should make unsub = noop, and then do an early return, no?